core/string/ustring.cpp

```diff
@@ -1302,6 +1302,43 @@ Vector<int> String::split_ints_mk(const Vector<String> &p_splitters, bool p_allo
 	return ret;
 }
 
+Vector<String> String::split_lines() const {
+	Vector<String> ret;
+	if (is_empty()) {
+		return ret;
+	}
+	int from = 0;
+	int end = length();
+	int current = 0;
+	while (current < end) {
+		char32_t chr = get_data()[current];
```

Comment:
```suggestion
	const char32_t *data = ptr();
	while (current < end) {
		char32_t chr = data[current];
```
Avoid fetching this every iteration, also no need to use `get_data` as we know the string is non-empty